app/src/main/java/com/everest/userinfo/validations/ValidateFields.kt

```diff
@@ -0,0 +1,65 @@
+package com.everest.userinfo.validations
+
+
+import android.util.Patterns
+import android.widget.Toast
+import com.everest.userinfo.HomeActivity
+import com.everest.userinfo.databinding.ActivityHomeBinding
+
+class ValidateFields constructor(
+    private var binding: ActivityHomeBinding,
+    private var homeActivity: HomeActivity
+) {
+
+    fun validateInputs(): Boolean {
+        if (!Patterns.EMAIL_ADDRESS.matcher(binding.inputEmail.text.trim()).matches()) {
+            Toast.makeText(
```

Comment:
we can pass the email and everything to validate, using views here again makes it tightly coupled to ActivityHomeBinding.